### PR TITLE
Preliminary support for the Daisy "Patch.Init" Eurorack module device

### PR DESCRIPTION
**Pull request by colinsullivan**
This adds a GUI class for the Daisy Patch.Init Eurorack module, based on the existing DaisyControlUI class.

The ifdefs are possibly looking a bit unwieldy, I'm down to help think through this more broadly.

The `KnobContext` et. al should be shared with DaisyControlUI.

Discussion welcome!

### CODE REVIEW

File: architecture/faust/gui/DaisyPatchInitControlUI.h [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+            fValue = fKey = fScale = "";
+        }
+    
+        void addCheckButton(const char* label, FAUSTFLOAT* zone)
+        {
+            if (fKey == "switch") {
+                std::unique_ptr<CheckButton> button = std::make_unique<CheckButton>(zone);
+                //if (fValue == "1") {
+                    //button->Init(fSeed->GetPin(SW_1_PIN), fRate);
+                //}
```
Comment (colinsullivan) [inline architecture/faust/gui/DaisyPatchInitControlUI.h:178]: TODO

File: architecture/faust/gui/DaisyPatchInitControlUI.h [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+        }
+    
+        // -- active widgets
+        void addButton(const char* label, FAUSTFLOAT* zone)
+        {
+            if (fKey == "switch") {
+                std::unique_ptr<SwitchButton> button = std::make_unique<SwitchButton>(zone);
+                //if (fValue == "1") {
+                    //button->Init(fSeed->GetPin(SW_1_PIN), fRate);
+                //}
```
Comment (colinsullivan) [inline architecture/faust/gui/DaisyPatchInitControlUI.h:166]: TODO

File: architecture/faust/gui/DaisyPatchInitControlUI.h [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+            bool fQuantizeToStep;
+            
+            AnalogKnob(daisy::AnalogControl* control, FAUSTFLOAT* zone, std::unique_ptr<ValueConverter>& converter, int rate, FAUSTFLOAT step, bool quantize)
+            : UpdatableZone(zone), fConverter(std::move(converter)), fControl(control), fStep(step), fQuantizeToStep(quantize)
+            {
+            }
+            
+            void update()
+            {
+                FAUSTFLOAT newValue = fConverter->ui2faust(fControl->Process());
```
Comment (colinsullivan) [inline architecture/faust/gui/DaisyPatchInitControlUI.h:80]: Unlike inside `DaisyControlUI`, no ADC related calls are here, the `DaisyPatchSM` class exposes these "control" instances already.

Much of the other code is the same and could be shared between them though.

**Review by colinsullivan — COMMENTED** [review]
A few thoughts...

**Comment by sletz** [comment]
@colinsullivan What is the status of this PR ?

**Comment by colinsullivan** [comment]
Rebased just now.

**Pull request merged**